presto-jdbc/src/main/java/io/prestosql/jdbc/AbstractPrestoResultSet.java

```diff
@@ -125,10 +125,36 @@
     // Since January 1, 1900 UTC is still December 31, 1899 in other zones, we are adding a 1 year margin.
     private static final long START_OF_MODERN_ERA_SECONDS = java.time.LocalDate.of(1901, 1, 1).toEpochDay() * SECONDS_PER_DAY;
 
-    private static final TypeConversions TYPE_CONVERSIONS =
+    @VisibleForTesting
+    static final Map<String, Class<?>> DEFAULT_OBJECT_REPRESENTATION = ImmutableMap.<String, Class<?>>builder()
+            .put("decimal", BigDecimal.class)
+            .put("date", java.sql.Date.class)
+            .put("time", java.sql.Time.class)
+            .put("time with time zone", java.sql.Time.class)
+            .put("timestamp", java.sql.Timestamp.class)
+            .put("timestamp with time zone", java.sql.Timestamp.class)
+            .put("interval year to month", PrestoIntervalYearMonth.class)
+            .put("interval day to second", PrestoIntervalDayTime.class)
+            .put("map", Map.class)
+            .put("row", Row.class)
+            .build();
+
+    @VisibleForTesting
+    static final TypeConversions TYPE_CONVERSIONS =
             TypeConversions.builder()
-                    .add("array", PrestoArray.class, List.class, array -> asList((Object[]) array.getArray()))
-                    .add("row", Row.class, Map.class, row -> {
+                    .add("decimal", String.class, BigDecimal.class, AbstractPrestoResultSet::parseBigDecimal)
+                    .add("date", String.class, Date.class, string -> parseDate(string, DateTimeZone.forID(ZoneId.systemDefault().getId())))
```

Comment:
can you replace `resultTimeZone` field with constant and reference the constant also here?

Reply:
`resultTimeZone` should not be constant because JVM time zone is settable